### PR TITLE
rootless: fill in correct storage conf default

### PR DESCRIPTION
**Pull request by giuseppe**
When the configuration file is specified, be sure to fill rootless
compatible values in the default configuration.

Closes: https://github.com/containers/libpod/issues/2510

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

File: libpod/runtime.go [hunk 1]
```diff
@@ -475,13 +475,32 @@ func NewRuntimeFromConfig(configPath string, options ...RuntimeOption) (runtime
 	// Set three fields not in the TOML config
 	runtime.config.StateType = defaultRuntimeConfig.StateType
 	runtime.config.OCIRuntime = defaultRuntimeConfig.OCIRuntime
-	runtime.config.StorageConfig = storage.StoreOptions{}
+
+	storageConf, err := util.GetDefaultStoreOptions()
+	if err != nil {
+		return nil, errors.Wrapf(err, "error retrieving rootless storage config")
```
Comment (TomSweeneyRedHat) [inline libpod/runtime.go]: would it always be rootless at this point?

Comment (rhatdan) [reply libpod/runtime.go]: Yes this should just say storage config.

File: libpod/runtime.go [hunk 2]
```diff
@@ -475,13 +475,32 @@ func NewRuntimeFromConfig(configPath string, options ...RuntimeOption) (runtime
...
 	runtime.config.TmpDir = tmpDir
+	if rootless.IsRootless() {
+		runtimeDir, err := util.GetRootlessRuntimeDir()
+		if err != nil {
+			return nil, err
+		}
+		// containers/image uses XDG_RUNTIME_DIR to locate the auth file.
+		// So make sure the env variable is set.
+		err = SetXdgRuntimeDir(runtimeDir)
+		if err != nil {
```
Comment (TomSweeneyRedHat) [inline libpod/runtime.go]: Think you could combine this and the previous line into one.

**Comment by TomSweeneyRedHat** [comment]
Looks like a test isn't buying this change.

**Comment by rhatdan** [comment]
LGTM other then the error message.

File: libpod/runtime.go (hunk 1 above)
Comment (giuseppe) [reply libpod/runtime.go]: fixed

File: libpod/runtime.go (hunk 2 above)
Comment (giuseppe) [reply libpod/runtime.go]: combined

**Pull request merged**